### ISSUE TITLE
Having `tsconfig.server.json` crashes SSR

### ISSUE THREAD

**Issue by Moeriki** [brillout/vite-plugin-ssr#3]
I have a seperate tsconfig for code in `server/` called `tsconfig.server.json`.

This crashes the SSR.

```js
    err: {
      "type": "Error",
      "message": "[vite-plugin-ssr][Wrong Usage] The file /tsconfig.server.json should be name `*.page.server.*`. Add the missing `.page`.",
      "stack":
          Error: [vite-plugin-ssr][Wrong Usage] The file /tsconfig.server.json should be name `*.page.server.*`. Add the missing `.page`.
              at /Users/moeriki/Projects/flex-template/node_modules/vite-plugin-ssr/user-files/getUserFiles.shared.ts:66:7
              at Array.forEach (<anonymous>)
              at /Users/moeriki/Projects/flex-template/node_modules/vite-plugin-ssr/user-files/getUserFiles.shared.ts:64:46
              at Array.forEach (<anonymous>)
              at assertExtensions (/Users/moeriki/Projects/flex-template/node_modules/vite-plugin-ssr/user-files/getUserFiles.shared.ts:59:34)
              at Object.getUserFiles (/Users/moeriki/Projects/flex-template/node_modules/vite-plugin-ssr/user-files/getUserFiles.shared.ts:20:3)
              at getPageIds (/Users/moeriki/Projects/flex-template/node_modules/vite-plugin-ssr/route.node.ts:141:21)
              at Object.route (/Users/moeriki/Projects/flex-template/node_modules/vite-plugin-ssr/route.node.ts:24:22)
              at render (/Users/moeriki/Projects/flex-template/node_modules/vite-plugin-ssr/render.node.ts:29:22)
    }
```

# Work around

Renamed `tsconfig.server.json` to `tsconfig.backend.json`.

Let me know if I should PR. 💛

**Comment by brillout** [comment]
> Let me know if I should PR. 💛

Sure 💛.

Let me know if you want guidance.

Checkout `CONTRIBUTING.md`.

**Comment by Moeriki** [comment]
I tested two possible solutions. I lack experience with rollup or vite plugins so I'm not sure if either of them makes sense.

- Change globs to explicitly check only `import.meta.glob('/**/*.page.server.*([a-zA-Z0-9])')`.
- Update `assertExtensions()` to filter the files with either an include or exclude list of file extensions.

See commits:

https://github.com/Moeriki/vite-plugin-ssr/commits/fix-ssr-server-page-error

**Comment by brillout** [comment]
The globs are purposely capturing `.server.*` files so that `vite-plugin-ssr` can warn the user in case he forgets the `.page.` middle suffix (it happened to me a couple of times).

Solution 2 should work! One thing though: we need to filter out the `.json` files from `userFiles`.

```ts
  assertExtensions(allUserFiles)

  const userFiles = Object.entries(allUserFiles[fileType]).map(
    ([filePath, loadFile]) => {
      return { filePath, loadFile }
    }
  )
```

Looking forward to see `vite-plugin-ssr`'s first PR land 😊

**Comment by brillout** [comment]
Fix released in `v0.1.0-beta.12`.
